internal/services/servicenetworking/application_load_balancer_association_resource.go

```diff
@@ -0,0 +1,227 @@
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9_.-]{0,62}[a-zA-Z0-9]$`), "the name must begin with a letter or number, end with a letter, number or underscore, and may contain only letters, numbers, underscores, periods, or hyphens. The value must be 1-64 characters long."),
+		},
+
+		"application_load_balancer_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: associationsinterface.ValidateTrafficControllerID,
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (t AssociationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (t AssociationResource) ModelObject() interface{} {
+	return &AssociationModel{}
+}
+
+func (t AssociationResource) ResourceType() string {
+	return "azurerm_application_load_balancer_association"
+}
+
+func (t AssociationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return associationsinterface.ValidateAssociationID
+}
+func (t AssociationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			trafficControllerClient := metadata.Client.ServiceNetworking.TrafficControllerInterface
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			var config AssociationModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			albId, err := trafficcontrollerinterface.ParseTrafficControllerID(config.ApplicationLoadBalancerId)
+			if err != nil {
+				return err
+			}
+
+			id := associationsinterface.NewAssociationID(albId.SubscriptionId, albId.ResourceGroupName, albId.TrafficControllerName, config.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of exisiting %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(t.ResourceType(), id)
+			}
+
+			controller, err := trafficControllerClient.Get(ctx, *albId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *albId, err)
+			}
+
+			association := associationsinterface.Association{
+				Properties: &associationsinterface.AssociationProperties{
+					Subnet: &associationsinterface.AssociationSubnet{
+						Id: config.SubnetId,
+					},
+					AssociationType: associationsinterface.AssociationTypeSubnets,
+				},
+			}
+
+			if controller.Model != nil {
+				association.Location = location.Normalize(controller.Model.Location)
```

Comment:
Is the location here required as it typically should be the same as its parent resource? If that is the case, then we can fail early (prior to sending the requset) with telling the user that the required location from the parent model is not retrieved.